src/modules/hip/kernel/lens_correction.hpp

```diff
@@ -0,0 +1,119 @@
+    float zCamera = id_y * inverseMatrix.f1[7] + inverseMatrix.f1[8] + id_x * inverseMatrix.f1[6];
+    float z = 1./zCamera, x = xCamera * z, y = yCamera * z;
+    float xSquare = x * x, ySquare = y * y;
+    float r2 = xSquare + ySquare, xyMul2 = 2 * x * y;
+    float kr = (1 + ((rCoeff[2] * r2 + rCoeff[1]) * r2 + rCoeff[0]) * r2) / (1 + ((rCoeff[5] * r2 + rCoeff[4]) * r2 + rCoeff[3]) *r2);
+    float colLoc = fx * (x * kr + tCoeff[0] * xyMul2 + tCoeff[1] * (r2 + 2 * xSquare)) + u0;
+    float rowLoc = fy * (y * kr + tCoeff[0] * (r2 + 2 * ySquare) + tCoeff[1] * xyMul2) + v0;
+    *colRemapTableTemp = colLoc;
+    *rowRemapTableTemp = rowLoc;
+}
+
+// -------------------- Set 3 - Kernel Executors --------------------
+
+RppStatus hip_exec_lens_correction_tensor(RpptDescPtr dstDescPtr,
+                                          Rpp32f *rowRemapTable,
+                                          Rpp32f *colRemapTable,
+                                          RpptDescPtr remapTableDescPtr,
+                                          Rpp32f *cameraMatrix,
+                                          Rpp32f *distanceCoeffs,
+                                          Rpp32f *newCameraMatrix,
+                                          RpptROIPtr roiTensorPtrSrc,
+                                          RpptRoiType roiType,
+                                          rpp::Handle& handle)
+{
+    if (roiType == RpptRoiType::LTRB)
+        hip_exec_roi_converison_ltrb_to_xywh(roiTensorPtrSrc, handle);
+
+    int globalThreads_x = dstDescPtr->w;
+    int globalThreads_y = dstDescPtr->h;
+    int globalThreads_z = handle.GetBatchSize();
+
+    float *inverseMatrix = handle.GetInitHandle()->mem.mgpu.scratchBufferHip.floatmem;
+    hipLaunchKernelGGL(get_inverse_hip,
+                       dim3(1, 1, ceil((float)globalThreads_z/LOCAL_THREADS_Z)),
+                       dim3(1, 1, LOCAL_THREADS_Z),
+                       0,
+                       handle.GetStream(),
+                       (d_float9 *)cameraMatrix,
+                       (d_float9 *)inverseMatrix);
+
+    hipLaunchKernelGGL(compute_remap_tables_hip_tensor,
+                       dim3(ceil((float)globalThreads_x/LOCAL_THREADS_X), ceil((float)globalThreads_y/LOCAL_THREADS_Y), ceil((float)globalThreads_z/LOCAL_THREADS_Z)),
+                       dim3(LOCAL_THREADS_X, LOCAL_THREADS_Y, LOCAL_THREADS_Z),
+                       0,
+                       handle.GetStream(),
+                       rowRemapTable,
+                       colRemapTable,
+                       (d_float9 *)cameraMatrix,
+                       (d_float9 *)inverseMatrix,
+                       (d_float8 *)distanceCoeffs,
+                       (d_float9 *)newCameraMatrix,
+                       make_uint2(remapTableDescPtr->strides.nStride, remapTableDescPtr->strides.hStride),
+                       roiTensorPtrSrc);
+    hipDeviceSynchronize();
```

Comment:
Please remove deviceSynchronize. Stream synchronizes can be added between and after kernels